src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -5239,6 +5239,28 @@ public override bool MightContainExtensionMethods
 
         #endregion
 
+        internal void DiscoverInterceptors(ArrayBuilder<NamespaceOrTypeSymbol> toSearch)
+        {
+            foreach (var type in this.GetTypeMembers())
+            {
+                toSearch.Add(type);
+            }
+
+            if (!declaration.AnyMemberHasAttributes)
```

Comment:
@CyrusNajmabadi Trying to figure out how to make this check smarter. The declaration table doesn't include whole entries for non-type members. So perhaps it would make sense to add something to SingleTypeDeclaration to indicate which members might have 'InterceptsLocationAttribute' around the same time that 'AnyMemberHasAttributes' is [calculated](https://github.com/dotnet/roslyn/blob/815f0269ad689da732212fb9c0733220a1855569/src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs#L939).

Reply:
yes.  this check seems unnecessarily expensive.  why not update the decl table?

Reply:
I started to play with adding a BitVector to the SingleTypeDeclaration to track which members lexically may have the attribute or not. The work just started to balloon a little bit especially as the code which builds the set of member names is oriented around the green tree while the code that gets/checks QuickAttributes is oriented around the red tree. I do have the WIP lying around though so could take another stab at it.

Reply:
Punting out to #72410